src/aleph/http/client.clj

```diff
@@ -386,95 +392,314 @@
+                            (when log-level
+                              (.frameLogger builder (Http2FrameLogger. log-level)))
+                            (-> builder
+                                (.initialSettings (Http2Settings/defaultSettings))
+                                (.build)))
+
+        ;; For the client, this may never get used, since the server will rarely
+        ;; initiate streams, and PUSH_PROMISE is deprecated. Responses to client-
+        ;; initiated streams use a separate handler, though we *also* set that using
+        ;; h2-stream-chan-initializer. Regardless, Http2MultiplexHandler must be on
+        ;; the pipeline to get new outbound channels.
+        server-initiated-stream-chan-initializer
+        (AlephChannelInitializer.
+          (fn [^Channel ch]
+            (let [p (.pipeline ch)]
+              (.addLast p
+                        "client-inbound-handler"
+                        ^ChannelHandler (netty/channel-inbound-handler
+                                          {:channel-active
+                                           ([_ ctx]
+                                            (log/error "Cannot currently handle server-initiated streams. Closing channel")
+                                            (netty/close ctx))})))))
+
+        multiplex-handler
+        (Http2MultiplexHandler. server-initiated-stream-chan-initializer)]
+
+    (-> pipeline
+        (common/attach-idle-handlers idle-timeout)
+        (.addLast "http2-frame-codec" http2-frame-codec)
+        (.addLast "multiplex" multiplex-handler)
+        ;; TODO: add handler for conn-level frames?
+        (.addLast "ex-handler"
+                  ^ChannelHandler
+                  (netty/channel-inbound-handler
+                    {:exception-caught
+                     ([_ ctx ex]
+                      (log/error ex "Exception in HTTP/2 connection channel. Closing channel.")
+                      (netty/close ctx))})))
+
+    (log/debug "Conn chan pipeline:" pipeline)))
+
+(defn make-pipeline-builder
+  "Returns a function that initializes a new conn channel's pipeline.
+
+   Only adds universal handlers like SslHandler. Protocol-specific handlers
+   are added later, for ease of coordination with non-Netty setup.
+
+   Can't use an ApnHandler/ApplicationProtocolNegotiationHandler here,
+   because it's tricky to run Manifold code on Netty threads."
+  [{:keys [ssl? remote-address ssl-context]}]
+  (fn pipeline-builder
+    [^ChannelPipeline pipeline]
+    (when ssl?
+      (do
+        (.addLast pipeline
+                  "ssl-handler"
+                  (netty/ssl-handler (.channel pipeline) ssl-context remote-address))
+        (.addLast pipeline
+                  "pause-handler"
+                  ^ChannelHandler (netty/pause-handler))))))
 
 (defn close-connection [f]
   (f
     {:method :get
-     :url "http://example.com"
-     ::close true}))
+     :url    "http://example.com"
+     :aleph/close true}))
```

Comment:
Since this got modified, maybe we can remove the `::close` line 1090.

Reply:
Good eye, but that might break user code on the off-chance they did used the old key manually. Pretty unlikely, but I figured just in case...